client/inbox-panel/dissmiss-all-modal.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * External dependencies
+ */
+import { Button, Modal } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+const DissmissAllModal = ( { setShowDismissAllModal, dismissAllNotes } ) => {
+	const title = __( 'Dismiss all messages', 'woocommerce-admin' );
+	const message = __(
+		'Are you sure? Inbox messages will be dismissed forever.',
+		'woocommerce-admin'
+	);
+	const dismissActionText = __( 'Cancel', 'woocommerce-admin' );
+	const acceptActionText = __( 'Yes, dismiss all', 'woocommerce-admin' );
+	return (
+		<>
+			<Modal
+				title={ title }
+				className="woocommerce-inbox-dismiss-all-modal"
+				onRequestClose={ () => {
+					setShowDismissAllModal( false );
```

Comment:
Instead of passing props which give knowledge to the child about the parent, could we updated this to `onClose` and keep the props more agnostic?

Then the parent can handle its own logic accordingly:

```js
				<DismissAllModal
					onClose={ () => setShowDismissAllModal( false ) }
				/>
```

Reply:
Thank you for the suggestion. That's a good one. Updated in https://github.com/woocommerce/woocommerce-admin/pull/7871/commits/0c8ac9cff96d25ee6b939f3857ecc618988ab683